aspnetcore/signalr/Configuration.md

```diff
@@ -0,0 +1,124 @@
+
+MessagePack is a binary serialization format that provides serialization options for server and client. 
+
+```csharp
+using Microsoft.Extensions.DependencyInjection
+
+var connection = new HubConnectionBuilder()
+    .WithUrl("...")
+    .AddJsonHubProtocol(options => {
+        // options.PayloadSerializerSettings is a JSON.NET JsonSerializerSettings used to 
+        // configure how the payload is serialized
+        // Example: PascalCase JSON (camelCase by default)
+        options.PayloadSerializerSettings.ContractResolver = new DefaultContractResolver();
+    })
+    .Build();
+```
+
+### [HubConnection](https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs#L66) options
+
+The following table describes the `HubConnection` options for configuring the client:
+
+| Option | Description |
+| ------ | ----------- |
+| `ServerTimeout` | If the server doesn't respond within this time interval, the connection closes |
+| `HandshakeTimeout`  | If the server doesn't respond to the initial handshake message within this time interval, the connection closes |
+| `PingInterval` (added in 2.2) | If the client hasn't sent a message within this interval, a ping message is sent automatically to keep the connection open |
+
+### [HubOptions](https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.SignalR.Core/HubOptions.cs)
+
+The following table describes the `HubOptions` options for configuring the hub:
+
+| Option | Description |
+| ------ | ----------- |
+| `HandshakeTimeout` | If the client doesn't send an initial handshake message within this time interval, the connection will be closed |
+| `KeepAliveInterval` | If the server hasn't sent a message within this interval, a ping message will is sent automatically to keep the connection open |
+| `SupportedProtocols` | Protocols supported by this hub. For example, disabling JSON or MessagePack for certain hubs |
+| `EnableDetailedErrors` | If true, sends detailed error messages to the client when exceptions occur. This may contain sensitive data and is `false` by default |
+
+The `HubOptions` are configured in the `ConfigureServices` method of the `Startup` class, as shown in the following code sample:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddSignalR(hubOptions =>
+    {
+        hubOptions.EnableDetailedErrors = true;
+        hubOptions.KeepAliveInterval = TimeSpan.FromMinutes(1);
+    })
+}
+```
+
+Strongly-typed [HubOptions<T>](https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.SignalR.Core/HubOptions.cs) can be configured as follows: 
+
+```csharp
+  services.AddSignalR().AddHubOptions<HubName>(options =>
+  {
+      options.EnableDetailedErrors = true;
+  }
+```
+
+### HttpConnectionDispatcherOptions
+
+The following table describes the `HttpConnectionDispatcherOptions` options:
```

Comment:
Options related to the transport layer. Use these to restrict the transports that can be used by SignalR clients, as well as to configure advanced settings related to memory buffer management.

These options are configured by passing a delegate to `.MapHub<T>`.